### PR TITLE
Make internal endpoints explicitly internal instead of versioned

### LINKED ISSUE #5791: Make internal endpoints explicitly internal instead of versioned
Tasks:
- Remove versioning from the frontend API, e.g. `/api/v3/nr_measurements` -> `/api/internal/nr_measurements`
- Retain versioning for the public endpoints as documented in `api.md`
- Update docs.

### PR DESCRIPTION
**Pull request by wkoot**
Closes #5791

### CODE REVIEW

File: components/api_server/src/routes/metric.py [hunk 1]
```diff
@@ -4,22 +4,20 @@
...
 from shared.database.measurements import insert_new_measurement, latest_measurement, latest_successful_measurement
 from shared.model.metric import Metric
 from shared.utils.type import MetricId, SubjectId, Value
 from shared_data_model import DATA_MODEL
-
-from database.reports import insert_new_report, latest_report_for_uuids, latest_reports
-from model.actions import copy_metric, move_item
-from model.defaults import default_metric_attributes
 from utils.functions import sanitize_html, uuid
-
```
Comment (fniessink) [inline components/api_server/src/routes/metric.py]: Why are the imports reorganised? And why didn't ruff complain?

File: components/api_server/src/routes/report.py [hunk 2]
```diff
@@ -76,6 +76,7 @@ def get_report(database: Database, report_uuid: ReportId | None = None):
     return {"ok": True, "reports": summarized_reports}
 
 
+@bottle.get("/api/internal/report/<report_uuid>/metric_status_summary", authentication_required=False)
 @bottle.get("/api/v3/report/<report_uuid>/metric_status_summary", authentication_required=False)
```
Comment (fniessink) [inline components/api_server/src/routes/report.py]: Why is this endpoint internal? It's not used by the frontend I think?

File: components/api_server/src/routes/report.py [hunk 3]
```diff
@@ -85,6 +86,7 @@ def get_report_metric_status_summary(database: Database, report: Report, report_
     return {"report_uuid": report_uuid, "title": summarized_report["title"], **summarized_report["summary"]}
 
 
+@bottle.post("/api/internal/report/import", permissions_required=[EDIT_REPORT_PERMISSION])
```
Comment (fniessink) [inline components/api_server/src/routes/report.py]: Why is this endpoint internal? It's not used by the frontend I think?

File: components/api_server/src/routes/report.py [hunk 4]
```diff
@@ -136,12 +138,14 @@ def post_report_copy(database: Database, report: Report, report_uuid: ReportId):
...
 
 
+@bottle.get("/api/internal/report/<report_uuid>/pdf", authentication_required=False)
 @bottle.get("/api/v3/report/<report_uuid>/pdf", authentication_required=False)
 def export_report_as_pdf(report_uuid: ReportId):
     """Download the report as PDF."""
     return export_as_pdf(report_uuid)
 
 
+@bottle.get("/api/internal/report/<report_uuid>/json", authentication_required=True)
```
Comment (fniessink) [inline components/api_server/src/routes/report.py]: This endpoint is not used by the frontend, so why make it internal?

File: components/api_server/src/routes/search.py [hunk 5]
```diff
@@ -75,6 +75,7 @@ def match_attribute(
     return attribute_name in domain_object and domain_object[attribute_name] == attribute_value
 
 
+@bottle.post("/api/internal/<domain_object_type>/search", authentication_required=False)
```
Comment (fniessink) [inline components/api_server/src/routes/search.py]: Why is this endpoint internal? It's not used by the frontend?

File: components/api_server/src/routes/server.py [hunk 6]
```diff
@@ -6,6 +6,7 @@
 QUALITY_TIME_VERSION = "5.8.0"
 
 
+@bottle.get("/api/internal/server", authentication_required=False)
```
Comment (fniessink) [inline components/api_server/src/routes/server.py]: Why is this endpoint internal? It's not used by the frontend I think?

File: components/collector/tests/source_collectors/quality_time/test_metrics.py [hunk 7]
```diff
@@ -13,7 +13,7 @@ class QualityTimeMetricsTest(QualityTimeTestCase):
     def setUp(self):
         """Set up test data."""
         super().setUp()
-        self.api_url = f"{self.url}/api/v3/report"
+        self.api_url = f"{self.url}/api/internal/report"
```
Comment (fniessink) [inline components/collector/tests/source_collectors/quality_time/test_metrics.py]: I think the Quality-time collector should use the public API. In theory the source Quality-time could be a different version so it's better to go via a versioned endpoint.

File: components/collector/tests/source_collectors/quality_time/test_source_up_to_dateness.py [hunk 8]
```diff
@@ -18,7 +18,7 @@ class QualityTimeSourceUpToDatenessTest(QualityTimeTestCase):
     def setUp(self):
         """Set up test data."""
         super().setUp()
-        self.api_url = f"{self.url}/api/v3/reports"
+        self.api_url = f"{self.url}/api/internal/reports"
```
Comment (fniessink) [inline components/collector/tests/source_collectors/quality_time/test_source_up_to_dateness.py]: I think the Quality-time collector should use the public API. In theory the source Quality-time could be a different version so it's better to go via a versioned endpoint.

File: docs/src/api.md
Comment (fniessink) [inline docs/src/api.md:1]: The file is reformatted, but not changed? I can't tell. Please move the reformatting to a different PR.

File: components/api_server/src/routes/auth.py [hunk 9]
```diff
@@ -127,6 +127,7 @@ def verify_user(database: Database, username: str, password: str) -> User:
     return user
 
 
+@bottle.post("/api/internal/login", authentication_required=False)
 @bottle.post("/api/v3/login", authentication_required=False)
```
Comment (fniessink) [inline components/api_server/src/routes/auth.py:131]: Why both internal and v3?

File: components/api_server/src/routes/auth.py [hunk 10]
```diff
@@ -145,13 +146,15 @@ def login(database: Database) -> dict[str, bool | str]:
     }
 
 
+@bottle.post("/api/internal/logout", authentication_required=True)
 @bottle.post("/api/v3/logout", authentication_required=True)
```
Comment (fniessink) [inline components/api_server/src/routes/auth.py:150]: Why both internal and v3?

File: components/api_server/src/routes/auth.py [hunk 11]
```diff
@@ -145,13 +146,15 @@ def login(database: Database) -> dict[str, bool | str]:
...
 def logout(database: Database) -> dict[str, bool]:
     """Log the user out."""
     delete_session(database)
     return {"ok": True}
 
 
+@bottle.get("/api/internal/public_key", authentication_required=False)
 @bottle.get("/api/v3/public_key", authentication_required=False)
```
Comment (fniessink) [inline components/api_server/src/routes/auth.py]: This endpoint is not used by the frontend, so why internal?

**Review by fniessink — CHANGES_REQUESTED** [review]
I think we need to consider more carefully which endpoints are internal, which are public, and which are both. I see many endpoints marked as internal that I think are not used by the frontend. I also see endpoints marked as public that I think are only internal.

Also, since we're changing the API, I think these changes should be added to the changelog.

File: components/api_server/src/routes/metric.py (hunk 1 above)
Comment (wkoot) [reply components/api_server/src/routes/metric.py]: Oh, that's weird! Probably incorrect IDE autoformatting, or perhaps it was even ruff reformatting it when I put it through `--fix`

Comment (wkoot) [reply components/api_server/src/routes/metric.py]: Just checked, ruff doesn't raise any objections in either way the imports are organised. Perhaps it thinks they might be external package imports also?

**Review by fniessink — APPROVED** [review]

**Pull request merged**